cmd/terramate/cli/cli.go

```diff
@@ -1328,23 +1329,55 @@ func (c *cli) printStacks() {
 	if err != nil {
 		c.fatal("Unable to list stacks", err)
 	}
-
 	c.gitFileSafeguards(false)
 
-	for _, entry := range c.filterStacks(report.Stacks) {
-		stack := entry.Stack
+	c.printStacksList(report.Stacks, c.parsedArgs.List.Why, c.parsedArgs.List.RunOrder)
+}
+
+func (c *cli) printStacksList(allStacks []stack.Entry, why bool, runOrder bool) {
+	filteredStacks := c.filterStacks(allStacks)
+
+	find := func(stackDir string) (stack.Entry, bool) {
```

Comment:
I don't get why this `find` is needed 🤔

Reply:
the `--why` handling needs a `reason`. The `reason` is only present in the `stack.Entry` type, not in the  `*config.Stack` type.

Reply:
A `find` is strange because you have the list at hand already and the elements are the same, just a different data structure.
What do you think of making a map of `stack.Dir -> reason` when building the config.List?

```
reasons = map[project.Path]string{}
stacks := make(config.List[*config.SortableStack], len(filteredStacks))
for i, entry := range filteredStacks {
	stacks[i] = entry.Stack.Sortable()
	reasons[entry.Stack.Dir] = entry.Reason
}
```

then later you do:
```
if why {
    printer.Stdout.Println(stdfmt.Sprintf("%s - %s", friendlyDir, reasons[st.Dir])
} else {
    ...
}
```

Reply:
That sounds great, just fixed it.